main/input_default.cpp

```diff
@@ -430,6 +437,7 @@ void InputDefault::_parse_input_event_impl(const Ref<InputEvent> &p_event, bool
 				Action action;
 				action.physics_frame = Engine::get_singleton()->get_physics_frames();
 				action.idle_frame = Engine::get_singleton()->get_idle_frames();
+				action.timestamp = OS::get_singleton()->get_ticks_usec();
```

Comment:
This still cannot work, you have to update the timestamp only when the pressed value changed. Otherwise, when only the strength changes (for axes), the duration will be modified too.

Reply:
Just tested, the timestamp is not modified whenever axes (strength) changes (only when it goes to 0 or 1). So works on my machine. :eyes:

See updated test project in the PR description.

Reply:
Ah sorry, I did not see the check for changing pressed status at the top. You are right :)